gobblin-core/src/main/java/gobblin/converter/json/JsonStringToJsonIntermediateConverter.java

```diff
@@ -39,39 +46,6 @@ public JsonArray convertSchema(String inputSchema, WorkUnitState workUnit) throw
       throws DataConversionException {
     JsonParser jsonParser = new JsonParser();
     JsonObject inputRecord = (JsonObject) jsonParser.parse(strInputRecord);
-    JsonObject outputRecord = new JsonObject();
-
-    for (int i = 0; i < outputSchema.size(); i++) {
-      String expectedColumnName = outputSchema.get(i).getAsJsonObject().get("columnName").getAsString();
-
-      if (inputRecord.has(expectedColumnName)) {
-        //As currently gobblin.converter.avro.JsonIntermediateToAvroConverter is not able to handle complex schema's so storing it as string
-
-        if (inputRecord.get(expectedColumnName).isJsonArray()) {
-          outputRecord.addProperty(expectedColumnName, inputRecord.get(expectedColumnName).toString());
-        } else if (inputRecord.get(expectedColumnName).isJsonObject()) {
-          //To check if internally in an JsonObject there is multiple hierarchy
-          boolean isMultiHierarchyInsideJsonObject = false;
-          for (Map.Entry<String, JsonElement> entry : ((JsonObject) inputRecord.get(expectedColumnName)).entrySet()) {
-            if (entry.getValue().isJsonArray() || entry.getValue().isJsonObject()) {
-              isMultiHierarchyInsideJsonObject = true;
-              break;
-            }
-          }
-          if (isMultiHierarchyInsideJsonObject) {
-            outputRecord.addProperty(expectedColumnName, inputRecord.get(expectedColumnName).toString());
-          } else {
-            outputRecord.add(expectedColumnName, inputRecord.get(expectedColumnName));
-          }
-
-        } else {
-          outputRecord.add(expectedColumnName, inputRecord.get(expectedColumnName));
-        }
-      } else {
-        outputRecord.add(expectedColumnName, JsonNull.INSTANCE);
-      }
-
-    }
-    return new SingleRecordIterable<>(outputRecord);
+    return Lists.newArrayList(inputRecord);
```

Comment:
I will ask Plaban about this. I really don't understand why all of this logic is there, from what I understand it is just replicating the same json object with some exceptions (like destroying arrays). If it is intentional, I'll just make a new converter.